File: src/python/PythonSDK/foundationallm/models/language_models/embedding_model.py
from typing import Optional
from pydantic import BaseModel
from foundationallm.models.language_models import LanguageModelType, LanguageModelProvider

class EmbeddingModel(BaseModel):
    """Embedding model metadata model."""
    type: str = LanguageModelType.OPENAI
    provider: Optional[str] = LanguageModelProvider.MICROSOFT
    deployment: Optional[str] = None
    model: Optional[str] = "text-embedding-ada-002"
    chunk_size: Optional[int] = 1000

File: src/dotnet/AgentFactory/Models/Orchestration/Metadata/EmbeddingModel.cs
using Newtonsoft.Json;

namespace FoundationaLLM.AgentFactory.Core.Models.Orchestration.Metadata
{
    /// <summary>
    /// 
    /// </summary>
    public class EmbeddingModel
    {
        /// <summary>
        /// The type of the embedding model.
        /// </summary>
        [JsonProperty("type")]
        public string? Type { get; set; }

        /// <summary>
        /// The provider of the emberdding model.
        /// </summary>
        [JsonProperty("provider")]
        public string? Provider { get; set; }

        /// <summary>
        /// Name of the embedding model deployment.
        /// </summary>
        [JsonProperty("deployment")]
        public string? Deployment { get; set; }

        /// <summary>
        /// Name of the embedding model.
        /// </summary>
        [JsonProperty("model")]
        public string? Model { get; set; }

        /// <summary>
        /// The chunk size to use when creating vectors from the embedding model.
        /// </summary>
        [JsonProperty("chunk_size")]
        public int ChunkSize { get; set; } = 1000;
    }
}
